docs/meshstack.cloudfoundry.metering.md

```diff
@@ -12,34 +12,76 @@ meshStack supports metering and billing for all versions of Cloud Foundry.
 
 ## Supported Resources
 
-Resources with the following traits are addressable in the Product Catalog.
+Resources with the following traits are addressable in the [Product Catalog](meshstack.billing-configuration.md#defining-a-custom-product-catalog). Operators can use the traits of these resources to define fine-granular product and pricing rules.
 
-### Apps
+### Service Instance
+<!--snippet:mesh.kraken.productcatalog.traits.cloudfoundry.service.instance#type-->
 
-```text
-id: cf.app
-traits:
- - SIZE (Instances * RAM)
- - EVENT
- - APP_STATE
-```
 
-### Marketplace
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
```

Comment:
d: I am wondering whether this dhall format is confusing, as the product catalog cannot be defined via dhall. And not sure whether it will in future. Putting the maintenance of the product catalog into the application is also a valid option as an alternative to dhall. That way admin could maintain the catalog by themself. But for sure, changing the catalog can be complex and may not be feasible for admins at the customer, but we'd have to think this through in the end which option fits better.

So for documenting it now like this, it is good to have the traits documented, but i am just not sure about the dhall format as it might bring readers into a wrong direction.

Reply:
I'm also not 100% sure if dhall is going to be the "primary" configuration format for the product catalog. But that's a decision I did not want to force at this time with this PR. My primary motivation was to get something (semi-)"automated" so we can generate the proper trait definitions and documentation of their values from meshStack's source code without (much) manual effort. The easiest path I saw to that goal was to reuse the existing dhall snippet infrastructure. 

Some additional thoughts: we will very likely also expose the product catalog via the API and in that case going from dhall -> YML -> meshObject API becomes very feasible, with the added advantage of having proper type definitions and ways to auto-generate catalogs with dhall. Of course we could also generate asciidocs like with the API... but that wasn't the obvious choice for me here either

Reply:
ok, sounds reasonable and especially the semi-automatic generation at the moment is a nice thing to have. So i am fine with that approach for now.